### DISCUSSION TITLE
Response: {error:invalid_grant,error_description:Invalid user credentials}

### DISCUSSION THREAD

**Discussion by mjonline99** [keycloak/keycloak#15210] [Q&A]
hello,
im getting this message while login to keycloak, can somebody help me with this error.

Response: {error:invalid_grant,error_description:Invalid user credentials}

![Code01](https://user-images.githubusercontent.com/116884466/198571246-7a68de66-4f04-4b3d-bc52-6f5c8c211e6b.png)

**Comment by kkcmadhu** [comment]
does your client (rest-client) has Direct Access Grants Enabled/ basic auth enabled?
also you may have to pass client secret if your client is not a public client..

**Reply by mjonline99** [reply]
does your client (rest-client) has Direct Access Grants Enabled/ basic auth enabled?
>> can you pls guide me how to check this?

also you may have to pass client secret if your client is not a public client..
>> how to do this?

**Reply by kkcmadhu** [reply]
![image](https://user-images.githubusercontent.com/997582/198580436-9684993a-1229-4587-bc40-1b899c1259ed.png)
refer documentation here
https://www.keycloak.org/docs/latest/server_admin/index.html#assembly-managing-clients_server_administration_guide

**Reply by mjonline99** [reply]
thank you...
i just verified it, both option as set as marked!
im accessing straight with a database application java script to keycloak...

**Reply by kkcmadhu** [reply]
invalid user credentails  -> check your password again, try resetting and give it a shot, not sure.

**Reply by mjonline99** [reply]
yes, it was working, thank you so much!
